### ISSUE TITLE
Updating ServerTech PDU device support 

### ISSUE THREAD

**Issue by papatango71** [librenms/librenms#8031]
DO NOT DELETE THIS INFORMATION.

> Please read this information carefully.

GitHub issues is for known/validated bugs, please do not post issues asking for help or how to do X, Y or Z. 

You can use our [Discord server](https://t.libren.ms/discord) to ask questions or our [community site](https://community.librenms.org).

If you have a feature request please post this on our [community site](https://community.librenms.org/c/feature-requests).

Please confirm each of the sections below by putting an `x` in the box like `[x]`.

- [X] Is your install up to date? [Updating your install](http://docs.librenms.org/General/Updating/)
      Please do not submit an issue if your install is not up to date within the last 24 hours or on a stable monthly release.
- [X] Please include all of the information between the `====================================` section of `./validate.php` which you can run from the cli.
- [X] Please provide ALL info asked for [here](http://docs.librenms.org/Support/FAQ/#faq20).
- [X] Please provide as much detail as possible.
- [X] Please do NOT post more than 10 lines of debug information here, use a pastebin service or GitHub Gists.

librenms@nms01-eqch2
[/opt/librenms]$ ./validate.php
====================================
Component | Version
--------- | -------
LibreNMS  | 1.33-304-ge16422a
DB Schema | 227
PHP       | 7.0.22-0ubuntu0.16.04.1
MySQL     | 10.0.31-MariaDB-0ubuntu0.16.04.2
RRDTool   | 1.5.5
SNMP      | NET-SNMP 5.7.3
====================================

[OK]    Database connection successful
[OK]    Database schema correct

DISCOVERY :: https://p.libren.ms/view/32314270
>> Error'ed out 
  Paste endpoint: https://p.libren.ms/api/create
  ifStackStatus: Unknown Object Identifier (Sub-id not found: (top) -> ifStackStatus)
  MIB search path: /opt/librenms/mibs:/opt/librenms/mibs/sentry
  Cannot find module (CPQHOST-MIB): At line 5 in (none)
  cpqHoFileSysEntry: Unknown Object Identifier (Sub-id not found: (top) -> cpqHoFileSysEntry)

POLLER :: https://p.libren.ms/view/9f0e3607

SNMPBULKWALK :: https://p.libren.ms/view/aca5e8a3

Senrty3.mib :: https://p.libren.ms/view/6652bb1a
Sentry3OIDTree.txt :: https://p.libren.ms/view/8bf2fe49

Sentry4.mib :: https://p.libren.ms/view/c2ff60a6
Sentry4OIDTree.txt :: https://p.libren.ms/view/c2d70970

**Comment by laf** [comment]
What needs updating?

**Comment by papatango71** [comment]
We have several ServerTech PDUs with sysObjectID .1.3.6.1.4.1.1718.3 and they show the Voltage & Current measurements.  Image attached.

![screen shot 2018-01-09 at 9 33 36 am](https://user-images.githubusercontent.com/29801388/34734318-34681b62-f520-11e7-871b-fc50c8607558.png)

We have begun adding the newer 3-phase PDUs from ServerTech with sysObjectID of .1.3.6.1.4.1.1718.4 and they are not showing the Voltage or Current measurements.  

![screen shot 2018-01-09 at 9 21 24 am](https://user-images.githubusercontent.com/29801388/34734307-237070d4-f520-11e7-9749-b07eb090f668.png)

I believe the issue is due to the OS.  The OS on the working PDUs is Sentry3 while the OS on the non-working PDUs is Sentry4. 

Below are the most up to date MIBs for both the Sentry3 and Sentry4.

Senrty3.mib :: https://p.libren.ms/view/6652bb1a
Sentry3OIDTree.txt :: https://p.libren.ms/view/8bf2fe49

Sentry4.mib :: https://p.libren.ms/view/c2ff60a6
Sentry4OIDTree.txt :: https://p.libren.ms/view/c2d70970

**Comment by dev9null** [comment]
I'm experiencing the same issue with our Sentry4 based CDUs -- only the Master_Cord value is retrieved.

Comparing the **includes/sentry3.yaml** and **includes/sentry4.yaml** there is an extra `discovery:` in the **sentry4.yaml** file. 

The version of the Sentry4 MIB in our librenms installation is 

```
   sentry4 MODULE-IDENTITY
        LAST-UPDATED "201611182340Z" -- 18 November 2016
        ORGANIZATION "Server Technology, Inc."
        CONTACT-INFO
            "Server Technology, Inc.
             1040 Sandhill Road
             Reno, NV 89521
             Tel: (775) 284-2000
             Fax: (775) 284-2065
             Email: mibmaster@servertech.com"
```

While the current Sentry4 MIB from STI's site is
(same as the one attached by papatango71)

```
    sentry4 MODULE-IDENTITY
        LAST-UPDATED "201706151830Z" -- 15 June 2017                                                                                                                                       
        ORGANIZATION "Server Technology, Inc."
        CONTACT-INFO
            "Server Technology, Inc.                                                                                                                                                       
             1040 Sandhill Road                                                                                                                                                            
             Reno, NV 89521                                                                                                                                                                
             Tel: (775) 284-2000                                                                                                                                                           
             Fax: (775) 284-2065                                                                                                                                                           
             Email: mibmaster@servertech.com"
```


```
root@librenms:/opt/librenms# ./validate.php 
====================================
Component | Version
--------- | -------
LibreNMS  | 1.37-21-g6c3473a
DB Schema | 239
PHP       | 7.0.25-0ubuntu0.16.04.1
MySQL     | 10.0.34-MariaDB-0ubuntu0.16.04.1
RRDTool   | 1.5.5
SNMP      | NET-SNMP 5.7.3
====================================
```

Device discovery: [https://p.libren.ms/view/2b1f265e](https://p.libren.ms/view/2b1f265e)
Device polling: [https://p.libren.ms/view/3dea683f](https://p.libren.ms/view/3dea683f)
SNMP bulk walk: [https://p.libren.ms/view/83dc74fc](https://p.libren.ms/view/83dc74fc)

**Comment by dev9null** [comment]
I added #8407 to my librenms install. Screen capture of one of the Sentry 4 PDUs with a temperature/humidity sensor attached.

![sentry4_pdu_with_temp_humidity](https://user-images.githubusercontent.com/7717928/37691708-f48023a4-2c70-11e8-91c2-06e6b44b2d89.png)


Looks mostly good. The voltages and current are correct. The temperature, however, is displayed a Celsius when it is Fahrenheit. It isn't 90 C in the hot row. It should show either 32.7 C or 90.8 F.

Let me know if y'all need sample data.

**Comment by Rosiak** [comment]
Implemented in #8407.
